### PR TITLE
Implement getOriginalException

### PR DESCRIPTION
**Pull request by alexbowers**

### CODE REVIEW

File: src/Framework/ExceptionWrapper.php [hunk 1]
```diff
@@ -64,6 +70,14 @@ public function getClassName()
     {
         return $this->className;
     }
+    
+    /**
+     * @return Throwable
+     */
+    public function getOriginalException()
+    {
+        return $this->originalException;
```
Comment (alexbowers) [inline src/Framework/ExceptionWrapper.php]: And this is why you don't use github to edit files. Fixing now.

**Comment by alexbowers** [comment]
@sebastianbergmann I'm not sure what here could have caused those tests to fail, since no regressions should have been introduced with this. Is this a widespread issue, or have I inadvertently done something bad?

**Comment by alexbowers** [comment]
Example usecase:

```php
public function run(TestResult $result = null)
{
    parent::run($result);

    if ($result->errorCount() > 0) {
        /**
         * @var $error PHPUnit_Framework_TestFailure
         */
        foreach ($result->errors() as $error) {
            $exception = $error->thrownException();
            if ($exception->getClassname() == ApiException::class) {
                $exception = $exception->getOriginalException();
                dump($exception->getErrors());
            }
        }
    }
}
```

This still causes phpunit to fail with the same exceptions, but instead of just saying: 

```
1) Tests\Product\CustomisationsTest::text_customisation
App\Exceptions\ApiException: Failed to pass validation

/var/www/sites/current/api.xxx.com/app/Exceptions/ApiException.php:27
/var/www/sites/current/api.xxx.com/app/Repositories/BaseRepository.php:135
/var/www/sites/current/api.xxx.com/app/Repositories/BaseRepository.php:127
/var/www/sites/current/api.xxx.com/app/Repositories/BaseRepository.php:207
/var/www/sites/current/api.xxx.com/tests/Product/CustomisationsTest.php:37
/var/www/sites/current/api.xxx.com/tests/TestCase.php:22
```

it now says:

```
PHPUnit 6.2-ga18fd3b by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)
Illuminate\Support\MessageBag {#167
  #messages: array:1 [
    "internal_additional_information.customisations.0" => array:1 [
      0 => "The internal additional information.customisations.0 field must be a valid customisation."
    ]
  ]
  #format: ":message"
}


Time: 1.81 seconds, Memory: 6.00MB

There was 1 error:

1) Tests\Product\CustomisationsTest::text_customisation
App\Exceptions\ApiException: Failed to pass validation

/var/www/sites/current/api.xxx.com/app/Exceptions/ApiException.php:27
/var/www/sites/current/api.xxx.com/app/Repositories/BaseRepository.php:135
/var/www/sites/current/api.xxx.com/app/Repositories/BaseRepository.php:127
/var/www/sites/current/api.xxx.com/app/Repositories/BaseRepository.php:207
/var/www/sites/current/api.xxx.com/tests/Product/CustomisationsTest.php:37
/var/www/sites/current/api.xxx.com/tests/TestCase.php:22

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

**Comment by alexbowers** [comment]
@sebastianbergmann Any chance I can get some feedback on this, and why it may have failed the tests?

**Comment by sebastianbergmann** [comment]
@alexbowers Please send a new pull request that fixes the test failure.

**Comment by alexbowers** [comment]
@sebastianbergmann Is there a guide anywhere on how to run the tests locally?

I get the following error:

```
PHP Fatal error:  Uncaught Error: Class 'PHPUnit\TextUI\Command' not found in /Users/alex/Development/phpunit/phpunit:53
Stack trace:
#0 {main}
  thrown in /Users/alex/Development/phpunit/phpunit on line 53

Fatal error: Uncaught Error: Class 'PHPUnit\TextUI\Command' not found in /Users/alex/Development/phpunit/phpunit:53
Stack trace:
#0 {main}
  thrown in /Users/alex/Development/phpunit/phpunit on line 53
```

**Comment by sebastianbergmann** [comment]
You need to install PHPUnit's dependencies first. Have a look at https://github.com/sebastianbergmann/phpunit/blob/master/.github/CONTRIBUTING.md#running-phpunits-own-test-suite

**Comment by alexbowers** [comment]
@sebastianbergmann That is the steps I followed, I still get the same error:

```

➜  Development git clone git://github.com/sebastianbergmann/phpunit.git
Cloning into 'phpunit'...
remote: Counting objects: 68369, done.
remote: Compressing objects: 100% (31/31), done.
remote: Total 68369 (delta 17), reused 25 (delta 9), pack-reused 68329
Receiving objects: 100% (68369/68369), 22.10 MiB | 7.13 MiB/s, done.
Resolving deltas: 100% (42492/42492), done.
➜  Development cd php
cd: no such file or directory: php
➜  Development cd phpunit
➜  phpunit git:(master) composer install
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 27 installs, 0 updates, 0 removals
  - Installing phar-io/version (1.0.1): Loading from cache
  - Installing phar-io/manifest (1.0.1): Loading from cache
  - Installing sebastian/recursion-context (3.0.0): Loading from cache
  - Installing doctrine/instantiator (1.0.5): Loading from cache
  - Installing sebastian/exporter (3.1.0): Loading from cache
  - Installing sebastian/diff (1.4.3): Loading from cache
  - Installing sebastian/comparator (2.0.0): Loading from cache
  - Installing webmozart/assert (1.2.0): Loading from cache
  - Installing phpdocumentor/reflection-common (1.0): Loading from cache
  - Installing phpdocumentor/type-resolver (0.2.1): Loading from cache
  - Installing phpdocumentor/reflection-docblock (3.1.1): Loading from cache
  - Installing phpspec/prophecy (v1.7.0): Loading from cache
  - Installing sebastian/object-reflector (1.1.1): Loading from cache
  - Installing sebastian/object-enumerator (3.0.2): Loading from cache
  - Installing sebastian/resource-operations (1.0.0): Loading from cache
  - Installing theseer/tokenizer (1.1.0): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/environment (3.0.3): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-token-stream (1.4.11): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/php-file-iterator (1.4.2): Loading from cache
  - Installing phpunit/php-code-coverage (5.2.1): Loading from cache
  - Installing phpunit/phpunit-mock-objects (4.0.1): Loading from cache
  - Installing myclabs/deep-copy (1.6.1): Loading from cache
  - Installing phpunit/php-timer (1.0.9): Loading from cache
  - Installing sebastian/global-state (2.0.0): Loading from cache
phpunit/php-code-coverage suggests installing ext-xdebug (^2.5.3)
sebastian/global-state suggests installing ext-uopz (*)
Writing lock file
Generating autoload files
➜  phpunit git:(master) ./phpunit
PHP Fatal error:  Uncaught Error: Class 'PHPUnit\TextUI\Command' not found in /Users/alex/Development/phpunit/phpunit:53
Stack trace:
#0 {main}
  thrown in /Users/alex/Development/phpunit/phpunit on line 53

Fatal error: Uncaught Error: Class 'PHPUnit\TextUI\Command' not found in /Users/alex/Development/phpunit/phpunit:53
Stack trace:
#0 {main}
  thrown in /Users/alex/Development/phpunit/phpunit on line 53

```

**Comment by alexbowers** [comment]
Ah. I have found the cause.

On my machine, I have a folder `~/Development` which deals with all of my side project coding stuff. That folder contains a folder called `vendor` which is from a past project.

The `array(__DIR__ . '/../../autoload.php', __DIR__ . '/../vendor/autoload.php', __DIR__ . '/vendor/autoload.php')` line finds this folder first, and this causes the issue.

It has the autoload.php, it autoloads it all in, but what it is loading is the incorrect vendor folder.

Is the order of these files here important, or should it not be `__DIR__./vendor/autoload.php` first?

If this package gets pulled into a different package, then it would not have its own vendor, which will cause it to not be found anyway.

**Comment by alexbowers** [comment]
@sebastianbergmann I have looked, and can't see how the tests are failing.

Is there any chance somebody with more knowledge of phpunits internals could have a look at them?

**Comment by alexbowers** [comment]
@sebastianbergmann Any chance I could get assistance with fixing the issues this had?

**Comment by alexbowers** [comment]
@sebastianbergmann Any chance I could get your assistance on this? Its a useful feature for us, and we have to currently use a fork since I couldn't resolve the issues you experienced.

**Comment by KamilKopaczyk** [comment]
Is there anything we can do to implement it?
It would be extremely useful for getting custom data from exceptions in TestListeners.

**Comment by denis-chmel** [comment]
@sebastianbergmann 
Please consider merging improved PR (all tests are passed now): #3012
It's not entirely clear why #2678 was failing Issue1351Test, but just turning $originalException to be private has helped. Well, I guess it's ok from OOP point of view (as long as there is a getter).

**Pull request merged**